assistant/assistant.go

```diff
@@ -236,6 +239,49 @@ func (assistant *Assistant) getPlugin(req request) response {
 	return response{SUCCESS, req.UUID, data}
 }
 
+type schema struct {
+	Types    map[string]interface{}
+	Defaults interface{}
```

Comment:
Shouldn't defaults also be a ```map[string]interface{}```?

Reply:
yes.

Reply:
I think we were just unsure how to convert a plugin to a map[string]interface{}, and it was easier to just use interface{} and let json.Marshal handle the stringifying.

Reply:
Also, if you look at lines 273-277 we tried to convert it like you said, but for some reason when we do a simple cast from a plugin struct to `map[string]interface{}` go does not like it. It was a lot simpler to plug in the plugin struct.

Reply:
Oh, so what you're essentially doing is just creating a blank instance of a plugin, putting it as the value of the ```Defaults``` field and then marshalling the entire schema object? Seems a little hacky as it although it kind of works, it isn't very clear as to what's going on. And when you do that, the marshal is going to include every field, even ones that are blank, which doesn't make much sense. 

To convert the plugin to a ```map[string]interface{}```, can't you loop through each of its fields using reflection and if the field as a value (eg: is not null or length 0 or something), then just add it to the map? 

Also you can't cast a struct in that way. You'd have to do it how I mentioned above

Reply:
Sure! A quick copying of getTypes fn should do it.